code/modules/medical_machinery/machines/advanced_medical_machinery.dm

```diff
@@ -0,0 +1,81 @@
+/obj/machinery/medical/dialysis
+	name = "Direct Blood Filtration Unit"
+	desc = "Automatically filtrates your blood from all chemicals."
+	icon_state = "dialysis"
+	///Amount of purged chems per process
+	var/purge_amount = 3
+
+/obj/machinery/medical/dialysis/RefreshParts()
+	for(var/obj/item/stock_parts/capacitor/C in component_parts)
+		purge_amount *= C.rating
```

Comment:
This will keep growing the purge amount infinitely. There are two ways to fix this

1) Set purge_amount to the initial purge amount before checking parts
```suggestion
/obj/machinery/medical/dialysis/RefreshParts()
	purge_amount = initial(purge_amount)
	for(var/obj/item/stock_parts/capacitor/C in component_parts)
		purge_amount *= C.rating
```

Reply:
2) Use a var to contain the new rating and set purge amount to that instead
```suggestion
/obj/machinery/medical/dialysis/RefreshParts()
	var/new_purge = initial(purge_amount)
	for(var/obj/item/stock_parts/capacitor/C in component_parts)
		new_purge *= C.rating
	purge_amount = new_purge
```